### PR TITLE
Add cashup feature

### PR DESCRIPTION
**Pull request by daN4cat**
The idea of this module is to mainly keep track of the cash in your drawer at opening and closing time.
So the entry needs to be created first thing in the morning, stating the opening cash position.
If during the day cash is added/removed out of the sales it should be added to the in/out field in the form, and it's better to add a note in the description to say what was that about.

At closing time the closed cash line needs to be added and adding the other payments should give a total matching the total of the day taken the expenses.

The checkbox notes is to say whether the cash has banknotes, just a memory note.

### CODE REVIEW

**Comment by daN4cat** [comment]
I went ahead, I just want to push this to master as it works in its initially intended form, if other people have ideas about improvement they are welcome to build on top of it.

File: application/migrations/sqlscripts/3.2.1_to_3.3.0.sql [hunk 1]
```diff
@@ -36,3 +36,48 @@ ALTER TABLE `ospos_sales_items`
 ALTER TABLE `ospos_receivings_items`
 	CHANGE COLUMN `discount_percent` `discount` DECIMAL(15,2) NOT NULL DEFAULT 0 AFTER `item_unit_price`,
 	ADD COLUMN `discount_type` TINYINT(2) NOT NULL DEFAULT '0' AFTER `discount`;
+
+
+--
+-- Add support for module cashups
+--
+
+INSERT INTO `ospos_modules` (`name_lang_key`, `desc_lang_key`, `sort`, `module_id`) VALUES
```
[the remark this answers is not in the document]
Comment (daN4cat) [reply application/migrations/sqlscripts/3.2.1_to_3.3.0.sql]: Actually I forgot to touch the date of the migration, I'll do that and commit. That's enough to get it to work.

[the remark this answers is not in the document]
Comment (daN4cat) [reply application/migrations/sqlscripts/3.2.1_to_3.3.0.sql]: done

File: application/migrations/sqlscripts/3.2.1_to_3.3.0.sql [hunk 2]
```diff
@@ -36,3 +36,48 @@ ALTER TABLE `ospos_sales_items`
...
+  `open_employee_id` int(10) NOT NULL,
+  `close_employee_id` int(10) NOT NULL,
+  `deleted` int(1) NOT NULL DEFAULT '0'
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Indexes for table `ospos_cash_up`
+
+ALTER TABLE `ospos_cash_up`
+  ADD PRIMARY KEY (`cashup_id`),
+  ADD KEY `open_employee_id` (`open_employee_id`),
```
[the remark this answers is not in the document]
Comment (daN4cat) [reply application/migrations/sqlscripts/3.2.1_to_3.3.0.sql]: Fixed

**Comment by daN4cat** [comment]
Two comments' fixes are now in master.
Also I found out that there was an issue with the Charset of the latest tables introduced with the previous version too. So fix all of that.
Anybody taking the latest code will get an updated migration file fixing everything and not facing issues.

**Comment by nathanzky** [comment]
I found an issue with cashup it is reflecting wrong amount on total. This happen when there is a due sale added. It shows correct cash amount but at the same time it deduct the total due amount to the total sale. So the close cash and the total cash is completely in correct. 
 
daily sales
cashup

**Comment by nathanzky** [comment]
I just modified some codes to add Due into the Cashup module in order to have correct cash on drawer.

cashup_update

**Comment by daN4cat** [comment]
Please submit the fix

**Comment by nathanzky** [comment]
Here are the codes I modify
application/controllers/Cashups.php
`<?php if ( ! defined('BASEPATH')) exit('No direct script access allowed');

require_once("Secure_Controller.php");

class Cashups extends Secure_Controller
{
	public function __construct()
	{
		parent::__construct('cashups');
	}

	public function index()
	{
		$data['table_headers'] = $this->xss_clean(get_cashups_manage_table_headers());

		// filters that will be loaded in the multiselect dropdown
		$data['filters'] = array('is_deleted' => $this->lang->line('cashups_is_deleted'));

		$this->load->view('cashups/manage', $data);
	}

	public function search()
	{
		$cash_up = 0;
		$search   = $this->input->get('search');
		$limit    = $this->input->get('limit');
		$offset   = $this->input->get('offset');
		$sort     = $this->input->get('sort');
		$order    = $this->input->get('order');
		$filters  = array(
					 'start_date' => $this->input->get('start_date'),
					 'end_date' => $this->input->get('end_date'),
					 'is_deleted' => FALSE);

		// check if any filter is set in the multiselect dropdown
		$filledup = array_fill_keys($this->input->get('filters'), TRUE);
		$filters = array_merge($filters, $filledup);
		$cash_ups = $this->Cashup->search($search, $filters, $limit, $offset, $sort, $order);
		$total_rows = $this->Cashup->get_found_rows($search, $filters);
		$data_rows = array();
		foreach($cash_ups->result() as $cash_up)
		{
			$data_rows[] = $this->xss_clean(get_cash_up_data_row($cash_up));
		}

		echo json_encode(array('total' => $total_rows, 'rows' => $data_rows));
	}

	public function view($cashup_id = -1)
	{
		$data = array();

		$data['employees'] = array();
		foreach($this->Employee->get_all()->result() as $employee)
		{
			foreach(get_object_vars($employee) as $property => $value)
			{
				$employee->$property = $this->xss_clean($value);
			}

			$data['employees'][$employee->person_id] = $employee->first_name . ' ' . $employee->last_name;
		}

		$cash_ups_info = $this->Cashup->get_info($cashup_id);

		foreach(get_object_vars($cash_ups_info) as $property => $value)
		{
			$cash_ups_info->$property = $this->xss_clean($value);
		}

		// open cashup
		if(empty($cash_ups_info->cashup_id))
		{
			$cash_ups_info->open_date = date('Y-m-d H:i:s');
			$cash_ups_info->close_date = $cash_ups_info->open_date;
			$cash_ups_info->open_employee_id = $this->Employee->get_logged_in_employee_info()->person_id;
			$cash_ups_info->close_employee_id = $this->Employee->get_logged_in_employee_info()->person_id;
		}
		// if all the amounts are null or 0 that means it's a close cashup
		elseif(floatval($cash_ups_info->closed_amount_cash) == 0 &&
			floatval($cash_ups_info->closed_amount_due) == 0 &&
			floatval($cash_ups_info->closed_amount_card) == 0 &&
			floatval($cash_ups_info->closed_amount_check) == 0)
		{
			// set the close date and time to the actual as this is a close session
			$cash_ups_info->close_date = date('Y-m-d H:i:s');

			// the closed amount starts with the open amount -/+ any trasferred amount
			$cash_ups_info->closed_amount_cash = $cash_ups_info->open_amount_cash + $cash_ups_info->transfer_amount_cash;

			// if it's date mode only and not date & time truncate the open and end date to date only
			if(empty($this->config->item('date_or_time_format')))
			{
				// search for all the payments given the time range
				$inputs = array('start_date' => substr($cash_ups_info->open_date, 0, 10), 'end_date' => substr($cash_ups_info->close_date, 0, 10), 'sale_type' => 'complete', 'location_id' => 'all');
			}
			else
			{
				// search for all the payments given the time range
				$inputs = array('start_date' => $cash_ups_info->open_date, 'end_date' => $cash_ups_info->close_date, 'sale_type' => 'complete', 'location_id' => 'all');
			}

			// get all the transactions payment summaries
			$this->load->model('reports/Summary_payments');
			$reports_data = $this->Summary_payments->getData($inputs);

			foreach($reports_data as $row)
			{
				if($row['payment_type'] == $this->lang->line('sales_cash'))
				{
					$cash_ups_info->closed_amount_cash += $this->xss_clean($row['payment_amount']);
				}
				elseif($row['payment_type'] == $this->lang->line('sales_due'))
				{
					$cash_ups_info->closed_amount_due += $this->xss_clean($row['payment_amount']);
				}
				elseif($row['payment_type'] == $this->lang->line('sales_debit') || 
						$row['payment_type'] == $this->lang->line('sales_credit'))
				{
					$cash_ups_info->closed_amount_card += $this->xss_clean($row['payment_amount']);
				}
				elseif($row['payment_type'] == $this->lang->line('sales_check'))
				{
					$cash_ups_info->closed_amount_check += $this->xss_clean($row['payment_amount']);
				}
			}

			// lookup expenses paid in cash
			$filters = array(
						 'only_cash' => TRUE,
						 'only_due' => FALSE,
						 'only_check' => FALSE,
						 'only_credit' => FALSE,
						 'only_debit' => FALSE,
						 'is_deleted' => FALSE);
			$payments = $this->Expense->get_payments_summary('', array_merge($inputs, $filters));

			foreach($payments as $row)
			{
				$cash_ups_info->closed_amount_cash -= $this->xss_clean($row['amount']);
			}

			$cash_ups_info->closed_amount_total = $this->_calculate_total($cash_ups_info->open_amount_cash, $cash_ups_info->transfer_amount_cash, $cash_ups_info->closed_amount_cash, $cash_ups_info->closed_amount_due, $cash_ups_info->closed_amount_card, $cash_ups_info->closed_amount_check);
		}

		$data['cash_ups_info'] = $cash_ups_info;

		$this->load->view("cashups/form", $data);
	}

	public function get_row($row_id)
	{
		$cash_ups_info = $this->Cashup->get_info($row_id);
		$data_row = $this->xss_clean(get_cash_up_data_row($cash_ups_info));

		echo json_encode($data_row);
	}

	public function save($cashup_id = -1)
	{
		$open_date = $this->input->post('open_date');
		$open_date_formatter = date_create_from_format($this->config->item('dateformat') . ' ' . $this->config->item('timeformat'), $open_date);

		$close_date = $this->input->post('close_date');
		$close_date_formatter = date_create_from_format($this->config->item('dateformat') . ' ' . $this->config->item('timeformat'), $close_date);

		$cash_up_data = array(
			'open_date' => $open_date_formatter->format('Y-m-d H:i:s'),
			'close_date' => $close_date_formatter->format('Y-m-d H:i:s'),
			'open_amount_cash' => parse_decimals($this->input->post('open_amount_cash')),
			'transfer_amount_cash' => parse_decimals($this->input->post('transfer_amount_cash')),
			'closed_amount_cash' => parse_decimals($this->input->post('closed_amount_cash')),
			'closed_amount_due' => parse_decimals($this->input->post('closed_amount_due')),
			'closed_amount_card' => parse_decimals($this->input->post('closed_amount_card')),
			'closed_amount_check' => parse_decimals($this->input->post('closed_amount_check')),
			'closed_amount_total' => parse_decimals($this->input->post('closed_amount_total')),
			'note' => $this->input->post('note') != NULL,
			'description' => $this->input->post('description'),
			'open_employee_id' => $this->input->post('open_employee_id'),
			'close_employee_id' => $this->input->post('close_employee_id'),
			'deleted' => $this->input->post('deleted') != NULL
		);

		if($this->Cashup->save($cash_up_data, $cashup_id))
		{
			$cash_up_data = $this->xss_clean($cash_up_data);

			//New cashup_id
			if($cashup_id == -1)
			{
				echo json_encode(array('success' => TRUE, 'message' => $this->lang->line('cashups_successful_adding'), 'id' => $cash_up_data['cashup_id']));
			}
			else // Existing Cashup
			{
				echo json_encode(array('success' => TRUE, 'message' => $this->lang->line('cashups_successful_updating'), 'id' => $cashup_id));
			}
		}
		else//failure
		{
			echo json_encode(array('success' => FALSE, 'message' => $this->lang->line('cashups_error_adding_updating'), 'id' => -1));
		}
	}

	public function delete()
	{
		$cash_ups_to_delete = $this->input->post('ids');

		if($this->Cashup->delete_list($cash_ups_to_delete))
		{
			echo json_encode(array('success' => TRUE, 'message' => $this->lang->line('cashups_successful_deleted') . ' ' . count($cash_ups_to_delete) . ' ' . $this->lang->line('cashups_one_or_multiple'), 'ids' => $cash_ups_to_delete));
		}
		else
		{
			echo json_encode(array('success' => FALSE, 'message' => $this->lang->line('cashups_cannot_be_deleted'), 'ids' => $cash_ups_to_delete));
		}
	}

	/*
	AJAX call from cashup input form to calculate the total
	*/
	public function ajax_cashup_total()
	{
		$open_amount_cash = parse_decimals($this->input->post('open_amount_cash'));
		$transfer_amount_cash = parse_decimals($this->input->post('transfer_amount_cash'));
		$closed_amount_cash = parse_decimals($this->input->post('closed_amount_cash'));
		$closed_amount_due = parse_decimals($this->input->post('closed_amount_due'));
		$closed_amount_card = parse_decimals($this->input->post('closed_amount_card'));
		$closed_amount_check = parse_decimals($this->input->post('closed_amount_check'));

		$total = $this->_calculate_total($open_amount_cash, $transfer_amount_cash, $closed_amount_due, $closed_amount_cash, $closed_amount_card, $closed_amount_check);

		echo json_encode(array('total' => to_currency_no_money($total)));
	}

	/*
	Calculate total
	*/
	private function _calculate_total($open_amount_cash, $transfer_amount_cash, $closed_amount_due, $closed_amount_cash, $closed_amount_card, $closed_amount_check)
	{
		return ($closed_amount_cash - $open_amount_cash - $transfer_amount_cash + $closed_amount_due + $closed_amount_card + $closed_amount_check);
	}
}
?>
`

**Comment by nathanzky** [comment]
@daN4cat can you guide me on how to submit the fixes?

These are the files I modified: 
application/controllers/Cashups.php
application/helpers/tabular_helper.php
application/language/en-US/cashups_lang.php
application/models/Cashup.php
application/views/cashups/form.php

**Comment by daN4cat** [comment]
Oh sorry I added a comment to the ticket but didn't see this.
Is your fix in a GitHub repo? I can pull that in for you if that's the case.

**Comment by nathanzky** [comment]
No its not in GitHub repo.. I made my fixes and test it on my local machine.

**Comment by daN4cat** [comment]
Ok, I'll look into that over the weekend.

**Comment by daN4cat** [comment]
@nathanzky can you please attach those 5 files here so I can patch the current code? Thanks

**Comment by nathanzky** [comment]
application/controllers/Cashups.php
`<?php if ( ! defined('BASEPATH')) exit('No direct script access allowed');

require_once("Secure_Controller.php");

class Cashups extends Secure_Controller
{
	public function __construct()
	{
		parent::__construct('cashups');
	}

	public function index()
	{
		$data['table_headers'] = $this->xss_clean(get_cashups_manage_table_headers());

		// filters that will be loaded in the multiselect dropdown
		$data['filters'] = array('is_deleted' => $this->lang->line('cashups_is_deleted'));

		$this->load->view('cashups/manage', $data);
	}

	public function search()
	{
		$cash_up = 0;
		$search   = $this->input->get('search');
		$limit    = $this->input->get('limit');
		$offset   = $this->input->get('offset');
		$sort     = $this->input->get('sort');
		$order    = $this->input->get('order');
		$filters  = array(
					 'start_date' => $this->input->get('start_date'),
					 'end_date' => $this->input->get('end_date'),
					 'is_deleted' => FALSE);

		// check if any filter is set in the multiselect dropdown
		$filledup = array_fill_keys($this->input->get('filters'), TRUE);
		$filters = array_merge($filters, $filledup);
		$cash_ups = $this->Cashup->search($search, $filters, $limit, $offset, $sort, $order);
		$total_rows = $this->Cashup->get_found_rows($search, $filters);
		$data_rows = array();
		foreach($cash_ups->result() as $cash_up)
		{
			$data_rows[] = $this->xss_clean(get_cash_up_data_row($cash_up));
		}

		echo json_encode(array('total' => $total_rows, 'rows' => $data_rows));
	}

	public function view($cashup_id = -1)
	{
		$data = array();

		$data['employees'] = array();
		foreach($this->Employee->get_all()->result() as $employee)
		{
			foreach(get_object_vars($employee) as $property => $value)
			{
				$employee->$property = $this->xss_clean($value);
			}

			$data['employees'][$employee->person_id] = $employee->first_name . ' ' . $employee->last_name;
		}

		$cash_ups_info = $this->Cashup->get_info($cashup_id);

		foreach(get_object_vars($cash_ups_info) as $property => $value)
		{
			$cash_ups_info->$property = $this->xss_clean($value);
		}

		// open cashup
		if(empty($cash_ups_info->cashup_id))
		{
			$cash_ups_info->open_date = date('Y-m-d H:i:s');
			$cash_ups_info->close_date = $cash_ups_info->open_date;
			$cash_ups_info->open_employee_id = $this->Employee->get_logged_in_employee_info()->person_id;
			$cash_ups_info->close_employee_id = $this->Employee->get_logged_in_employee_info()->person_id;
		}
		// if all the amounts are null or 0 that means it's a close cashup
		elseif(floatval($cash_ups_info->closed_amount_cash) == 0 &&
			floatval($cash_ups_info->closed_amount_due) == 0 &&
			floatval($cash_ups_info->closed_amount_card) == 0 &&
			floatval($cash_ups_info->closed_amount_check) == 0)
		{
			// set the close date and time to the actual as this is a close session
			$cash_ups_info->close_date = date('Y-m-d H:i:s');

			// the closed amount starts with the open amount -/+ any trasferred amount
			$cash_ups_info->closed_amount_cash = $cash_ups_info->open_amount_cash + $cash_ups_info->transfer_amount_cash;

			// if it's date mode only and not date & time truncate the open and end date to date only
			if(empty($this->config->item('date_or_time_format')))
			{
				// search for all the payments given the time range
				$inputs = array('start_date' => substr($cash_ups_info->open_date, 0, 10), 'end_date' => substr($cash_ups_info->close_date, 0, 10), 'sale_type' => 'complete', 'location_id' => 'all');
			}
			else
			{
				// search for all the payments given the time range
				$inputs = array('start_date' => $cash_ups_info->open_date, 'end_date' => $cash_ups_info->close_date, 'sale_type' => 'complete', 'location_id' => 'all');
			}

			// get all the transactions payment summaries
			$this->load->model('reports/Summary_payments');
			$reports_data = $this->Summary_payments->getData($inputs);

			foreach($reports_data as $row)
			{
				if($row['payment_type'] == $this->lang->line('sales_cash'))
				{
					$cash_ups_info->closed_amount_cash += $this->xss_clean($row['payment_amount']);
				}
				elseif($row['payment_type'] == $this->lang->line('sales_due'))
				{
					$cash_ups_info->closed_amount_due += $this->xss_clean($row['payment_amount']);
				}
				elseif($row['payment_type'] == $this->lang->line('sales_debit') || 
						$row['payment_type'] == $this->lang->line('sales_credit'))
				{
					$cash_ups_info->closed_amount_card += $this->xss_clean($row['payment_amount']);
				}
				elseif($row['payment_type'] == $this->lang->line('sales_check'))
				{
					$cash_ups_info->closed_amount_check += $this->xss_clean($row['payment_amount']);
				}
			}

			// lookup expenses paid in cash
			$filters = array(
						 'only_cash' => TRUE,
						 'only_due' => FALSE,
						 'only_check' => FALSE,
						 'only_credit' => FALSE,
						 'only_debit' => FALSE,
						 'is_deleted' => FALSE);
			$payments = $this->Expense->get_payments_summary('', array_merge($inputs, $filters));

			foreach($payments as $row)
			{
				$cash_ups_info->closed_amount_cash -= $this->xss_clean($row['amount']);
			}

			$cash_ups_info->closed_amount_total = $this->_calculate_total($cash_ups_info->open_amount_cash, $cash_ups_info->transfer_amount_cash, $cash_ups_info->closed_amount_cash, $cash_ups_info->closed_amount_due, $cash_ups_info->closed_amount_card, $cash_ups_info->closed_amount_check);
		}

		$data['cash_ups_info'] = $cash_ups_info;

		$this->load->view("cashups/form", $data);
	}

	public function get_row($row_id)
	{
		$cash_ups_info = $this->Cashup->get_info($row_id);
		$data_row = $this->xss_clean(get_cash_up_data_row($cash_ups_info));

		echo json_encode($data_row);
	}

	public function save($cashup_id = -1)
	{
		$open_date = $this->input->post('open_date');
		$open_date_formatter = date_create_from_format($this->config->item('dateformat') . ' ' . $this->config->item('timeformat'), $open_date);

		$close_date = $this->input->post('close_date');
		$close_date_formatter = date_create_from_format($this->config->item('dateformat') . ' ' . $this->config->item('timeformat'), $close_date);

		$cash_up_data = array(
			'open_date' => $open_date_formatter->format('Y-m-d H:i:s'),
			'close_date' => $close_date_formatter->format('Y-m-d H:i:s'),
			'open_amount_cash' => parse_decimals($this->input->post('open_amount_cash')),
			'transfer_amount_cash' => parse_decimals($this->input->post('transfer_amount_cash')),
			'closed_amount_cash' => parse_decimals($this->input->post('closed_amount_cash')),
			'closed_amount_due' => parse_decimals($this->input->post('closed_amount_due')),
			'closed_amount_card' => parse_decimals($this->input->post('closed_amount_card')),
			'closed_amount_check' => parse_decimals($this->input->post('closed_amount_check')),
			'closed_amount_total' => parse_decimals($this->input->post('closed_amount_total')),
			'note' => $this->input->post('note') != NULL,
			'description' => $this->input->post('description'),
			'open_employee_id' => $this->input->post('open_employee_id'),
			'close_employee_id' => $this->input->post('close_employee_id'),
			'deleted' => $this->input->post('deleted') != NULL
		);

		if($this->Cashup->save($cash_up_data, $cashup_id))
		{
			$cash_up_data = $this->xss_clean($cash_up_data);

			//New cashup_id
			if($cashup_id == -1)
			{
				echo json_encode(array('success' => TRUE, 'message' => $this->lang->line('cashups_successful_adding'), 'id' => $cash_up_data['cashup_id']));
			}
			else // Existing Cashup
			{
				echo json_encode(array('success' => TRUE, 'message' => $this->lang->line('cashups_successful_updating'), 'id' => $cashup_id));
			}
		}
		else//failure
		{
			echo json_encode(array('success' => FALSE, 'message' => $this->lang->line('cashups_error_adding_updating'), 'id' => -1));
		}
	}

	public function delete()
	{
		$cash_ups_to_delete = $this->input->post('ids');

		if($this->Cashup->delete_list($cash_ups_to_delete))
		{
			echo json_encode(array('success' => TRUE, 'message' => $this->lang->line('cashups_successful_deleted') . ' ' . count($cash_ups_to_delete) . ' ' . $this->lang->line('cashups_one_or_multiple'), 'ids' => $cash_ups_to_delete));
		}
		else
		{
			echo json_encode(array('success' => FALSE, 'message' => $this->lang->line('cashups_cannot_be_deleted'), 'ids' => $cash_ups_to_delete));
		}
	}

	/*
	AJAX call from cashup input form to calculate the total
	*/
	public function ajax_cashup_total()
	{
		$open_amount_cash = parse_decimals($this->input->post('open_amount_cash'));
		$transfer_amount_cash = parse_decimals($this->input->post('transfer_amount_cash'));
		$closed_amount_cash = parse_decimals($this->input->post('closed_amount_cash'));
		$closed_amount_due = parse_decimals($this->input->post('closed_amount_due'));
		$closed_amount_card = parse_decimals($this->input->post('closed_amount_card'));
		$closed_amount_check = parse_decimals($this->input->post('closed_amount_check'));

		$total = $this->_calculate_total($open_amount_cash, $transfer_amount_cash, $closed_amount_due, $closed_amount_cash, $closed_amount_card, $closed_amount_check);

		echo json_encode(array('total' => to_currency_no_money($total)));
	}

	/*
	Calculate total
	*/
	private function _calculate_total($open_amount_cash, $transfer_amount_cash, $closed_amount_due, $closed_amount_cash, $closed_amount_card, $closed_amount_check)
	{
		return ($closed_amount_cash - $open_amount_cash - $transfer_amount_cash + $closed_amount_due + $closed_amount_card + $closed_amount_check);
	}
}
?>
`

**Comment by HamidCh** [comment]
Is Day Beginning Cash Module is added to the OSPOS or not? If any help is needed I am available.

**Comment by HamidCh** [comment]
Is Day Beginning Cash Module is added to the OSPOS or not? If any help is needed I am available.

**Comment by daN4cat** [comment]
It's part of 3.3.0.

**Comment by MoiNKhulna** [comment]
Cashups-01

**Comment by HamidCh** [comment]
How can I access thi cashup feature?

**Comment by odiea** [comment]
https://bintray.com/jekkos/opensourcepos/download_file?file_path=opensourcepos.20190804203131.master.0a2894.zip

**Comment by HamidCh** [comment]
What about Credit Payment Feature?

**Comment by daN4cat** [comment]
@HamidCh not sure I follow. What is the credit payment implication with the daily cash and payment counting?

**Comment by HamidCh** [comment]
That is some other thread I am talking about. Is there any chance of adding Credit payment or DUE payment feature in this POS.

**Comment by daN4cat** [comment]
But this thread is about cashup, I struggle to see the point in mixing up things.

**Comment by tfjadrian** [comment]
Hello everyone!

Not sure if there is already, but I could not find a configuration option to control the cash up making, in that way that a restricted user (e. g. employee) can make a cash up only for him/her, and not for another user from the list.

(Thanks for all the people who make possible this pos software. I wish I had better programming skills, in order to contribute with you)

**Comment by daN4cat** [comment]
@tfjadrian I'm afraid there is no such option.

**Pull request merged**